tools/marvin/marvin/lib/vcenter.py

```diff
@@ -183,7 +192,157 @@ def get_clusters(self, dc, clus=None):
         """
         pass
 
+    def create_datacenter(self, dcname=None, service_instance=None, folder=None):
+        """
+        Creates a new datacenter with the given name.
+        Any % (percent) character used in this name parameter must be escaped,
+        unless it is used to start an escape sequence. Clients may also escape
+        any other characters in this name parameter.
+
+        An entity name must be a non-empty string of
+        less than 80 characters. The slash (/), backslash (\) and percent (%)
+        will be escaped using the URL syntax. For example, %2F
+
+        This can raise the following exceptions:
+        vim.fault.DuplicateName
+        vim.fault.InvalidName
+        vmodl.fault.NotSupported
+        vmodl.fault.RuntimeFault
+        ValueError raised if the name len is > 79
+        https://github.com/vmware/pyvmomi/blob/master/docs/vim/Folder.rst
+
+        Required Privileges
+        Datacenter.Create
+
+        :param folder: Folder object to create DC in. If None it will default to
+                       rootFolder
+        :param dcname: Name for the new datacenter.
+        :param service_instance: ServiceInstance connection to a given vCenter
+        :return:
+        """
+        if len(dcname) > 79:
+            raise ValueError("The name of the datacenter must be under "
+                             "80 characters.")
+
+        if folder is None:
+            folder = self.service_instance.content.rootFolder
+
+        if folder is not None and isinstance(folder, vim.Folder):
+            dc_moref = folder.CreateDatacenter(name=dcname)
```

Comment:
Hello @sanju1010 
Is it necessary to double check if folder is not None here? Also you mentioned some restrictions for the dcname, shouldnt they also raise and ValueError?